src/chatdlg.cpp

```diff
@@ -1071,7 +1071,8 @@ void ChatDlg::updateIsComposing(bool b)
 
 void ChatDlg::setChatState(ChatState state)
 {
-    if (PsiOptions::instance()->getOption("options.messages.send-composing-events").toBool() && (sendComposingEvents_ || (contactChatState_ != XMPP::StateNone))) {
+    bool sendAtStart = PsiOptions::instance()->getOption("options.messages.send-composing-events-at-start").toBool();
```

Comment:
I think you could assign this on line 132 and everything else is not needed

Reply:
maybe together with setChatState(XMPP::StateActive). And I think this new option has to be enabled by default according to standard and for consistency with other clients.

Reply:
sendComposingEvents_ seems to be used for legacy composing events, but setting the chat state according to the setting (and user caps) should do the trick. I didn't want to do it right away because this is my first look into the codebase, so I can't predict yet any possible side effects :)